src/libcrun/criu.c

```diff
@@ -218,6 +218,88 @@ libcrun_container_checkpoint_linux_criu (libcrun_container_status_t *status,
   return 0;
 }
 
+static int
+prepare_restore_mounts (runtime_spec_schema_config_schema *def,
+                        char *root,
+                        libcrun_error_t *err)
+{
+  int i;
+
+  /* Go through all mountpoints to be able to recreate missing mountpoints. */
+  for (i = 0; i < def->mounts_len; i++)
+    {
+      char *dest = def->mounts[i]->destination;
+      char *type = def->mounts[i]->type;
+      cleanup_close int root_fd = -1;
+      bool on_tmpfs = false;
+      int is_dir = 1;
+      size_t j;
+
+      /* cgroup restore should be handled by CRIU itself */
+      if (strcmp (type, "cgroup") == 0
+          || strcmp (type, "cgroup2") == 0)
+        continue;
+
+      for (j = 0; j < def->mounts_len; j++)
+        {
+          cleanup_free char *dest_loop = NULL;
+
+          xasprintf (&dest_loop, "%s/", def->mounts[j]->destination);
+          /* Check if the parent mount is on a tmpfs. CRIU restores
+           * all tmpfs. We do need to recreate directories on a tmpfs. */
+          if (strncmp (dest, dest_loop, strlen (dest_loop)) == 0 &&
+              strcmp (def->mounts[j]->type, "tmpfs") == 0)
```

Comment:
the tmpfs check need to be moved to right after `for (j = 0, ...`, i.e.

```c
for (j = 0; j < def->mounts_len; j++)
￼ {
￼ ￼ cleanup_free char *dest_loop = NULL;

￼ ￼ if strcmp (def->mounts[j]->type, "tmpfs") != 0)
￼ ￼ ￼ continue;
...
```
(and looks like `= NULL` will be needed in this case, so please ignore my earlier comment about it)

Reply:
Not sure I  understand what you mean with 'the tmpfs check need to be moved right after'.

Ah, maybe that if it is on a tmpfs I do not need to compare the paths at all...

Reply:
Well, I mean as I wrote it in the example above. The current code could have avoided allocating memory for dest_loop, the copy, and the comparison.